### PR TITLE
[camel-k] Camel-k deployment via our testsuite

### PR DESCRIPTION
**Pull request by mmuzikar**
//test: `@smoke`

### CODE REVIEW

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:90:47: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:37: 'integrations' hides a field. [HiddenField]
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:51: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:92:44: '{' is not preceded with whitespace. [WhitespaceAround]
utilities/src/main/java/io/syndesis/qe/bdd/
  CommonSteps.java:161:55: '{' is not preceded with whitespace. [WhitespaceAround]
utilities/src/main/java/io/syndesis/qe/camelk/
  CamelKTemplate.java:48:95: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
utilities/src/main/java/io/syndesis/qe/templates/
  SyndesisTemplate.java:267:51: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:37: 'integrations' hides a field. [HiddenField]
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:51: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
utilities/src/main/java/io/syndesis/qe/camelk/
  CamelKTemplate.java:48:95: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
utilities/src/main/java/io/syndesis/qe/templates/
  SyndesisTemplate.java:267:51: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:37: 'integrations' hides a field. [HiddenField]
ui-tests/src/test/java/io/syndesis/qe/steps/integrations/
  IntegrationSteps.java:91:51: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
utilities/src/main/java/io/syndesis/qe/camelk/
  CamelKTemplate.java:48:95: Unnecessary parentheses around lambda value. [UnnecessaryParentheses]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by mmuzikar** [comment]
@avano  this is mostly your code so you can have the pleasure of reviewing yourself :D

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

File: utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    }
+
+    public static void deployCamelKOperator() {
+        ProcessBuilder pb = new ProcessBuilder("docker", "pull", TestConfiguration.getCamelKOperator());
+
+        try {
+            pb.start().waitFor();
+        } catch (IOException | InterruptedException e) {
+            log.error("Unable to pull Camel K operator \"{}\"", TestConfiguration.getCamelKOperator(), e);
+        }
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java]: not needed?

File: utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    .map(line -> line.replaceAll("\\s+", "\t").split("\t"))
+                    .filter(parts -> parts.length == 3 && !"NAME".equals(parts[0]))
+                    .map(parts -> new CamelKIntegration(parts[0], parts[1], parts[2]))
+                    .collect(Collectors.toMap(CamelKIntegration::getId, Function.identity()));
+            }
+        } catch (Exception e) {
+            fail("Failed to list Camel K integrations", e);
+        }
+        return new HashMap<>();
+    }
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java]: isn't it possible to list the integrations as the integration `CR`?

File: utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    );
+    private static final String LOCAL_ARCHIVE_PATH = "/tmp/camelk.tar.gz";
+    private static final String LOCAL_ARCHIVE_EXTRACT_DIRECTORY = "/tmp/camelk";
+
+    public static void deploy() {
+        downloadArchive();
+        extractArchive();
+        installViaBinary();
+        OpenShiftUtils.binary().execute("secrets", "link", "camel-k-operator", "syndesis-pull-secret", "--for=pull");
+        OpenShiftUtils.binary().execute("delete", "pod", "-l", "name=camel-k-operator");
```
Comment (avano) [inline utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java]: I know its easier to do it with binary client, but the preferred way should be using the java client

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
@mmuzikar there is a new class for camel-k https://github.com/syndesisio/syndesis-qe/blob/master/utilities/src/main/java/io/syndesis/qe/resource/impl/CamelK.java

**Pull request closed without merging**